From cd7053cf724a5560287e912c867ad38eba83c89d Mon Sep 17 00:00:00 2001
From: FlameFlag <github@flameflag.dev>
Date: Sun, 12 Apr 2026 01:13:41 +0300
Subject: [PATCH 5/5] src: guard WebRTCDecoder usage with ENABLE_WEBRTC_VIDEO
 define

WebRTCDecoder.h includes metartc/yangplayer headers which are not
available. Add ENABLE_WEBRTC_VIDEO as a compile definition when the
feature is enabled, and guard all WebRTCDecoder includes and usage
with it in PrinterMgrView.cpp, SendToPrinter.cpp, and HttpServer.cpp.
---
 src/CMakeLists.txt                                 | 3 +++
 src/slic3r/GUI/HttpServer.cpp                      | 8 ++++----
 src/slic3r/GUI/print_manage/App/PrinterMgrView.cpp | 6 +++---
 src/slic3r/GUI/print_manage/App/SendToPrinter.cpp  | 4 ++--
 4 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ea0e33a7..38a00c63 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -19,6 +19,9 @@ add_subdirectory(hints)
 add_subdirectory(mcut)
 add_subdirectory(qoi)
 set(ENABLE_WEBRTC_VIDEO OFF)
+if(ENABLE_WEBRTC_VIDEO)
+    add_definitions(-DENABLE_WEBRTC_VIDEO)
+endif()
 add_subdirectory(video)
 # Adding libnest2d project for bin packing...
 add_subdirectory(libnest2d)
diff --git a/src/slic3r/GUI/HttpServer.cpp b/src/slic3r/GUI/HttpServer.cpp
index 1f3ac1da..87df12f4 100644
--- a/src/slic3r/GUI/HttpServer.cpp
+++ b/src/slic3r/GUI/HttpServer.cpp
@@ -5,10 +5,10 @@
 #include "slic3r/Utils/Http.hpp"
 #include "slic3r/Utils/NetworkAgent.hpp"
 #include <boost/regex.hpp>
-#if defined(__linux__) || defined(__LINUX__)
+#if (defined(__linux__) || defined(__LINUX__)) && defined(ENABLE_WEBRTC_VIDEO)
 #include "video/WebRTCDecoder.h"
-#include <boost/asio/steady_timer.hpp>
 #endif
+#include <boost/asio/steady_timer.hpp>
 
 #include "buildinfo.h"
 #if ENABLE_FFMPEG
@@ -172,7 +172,7 @@ void session::read_next_line()
                         else
                         {
                             video_url = (boost::format("http://%1%:8000/call/webrtc_local") % ip).str();
-                              #if defined(__linux__) || defined(__LINUX__)
+                              #if (defined(__linux__) || defined(__LINUX__)) && defined(ENABLE_WEBRTC_VIDEO)
                                 WebRTCDecoder::GetInstance()->startPlay(video_url); 
                               #endif
                         }
@@ -235,7 +235,7 @@ void session::sendFrame(const boost::system::error_code &ec,boost::asio::ip::tcp
             RTSPDecoder::GetInstance()->getFrameData(copy_frame);
             #endif
         }else{
-            #if defined(__linux__) || defined(__LINUX__)
+            #if (defined(__linux__) || defined(__LINUX__)) && defined(ENABLE_WEBRTC_VIDEO)
             copy_frame = WebRTCDecoder::GetInstance()->getFrameData();
             #endif
         }
diff --git a/src/slic3r/GUI/print_manage/App/PrinterMgrView.cpp b/src/slic3r/GUI/print_manage/App/PrinterMgrView.cpp
index 2cd03ddc..d230fa5a 100644
--- a/src/slic3r/GUI/print_manage/App/PrinterMgrView.cpp
+++ b/src/slic3r/GUI/print_manage/App/PrinterMgrView.cpp
@@ -39,7 +39,7 @@
 #include <boost/uuid/uuid.hpp>
 #include <boost/uuid/uuid_generators.hpp>
 #include <boost/uuid/uuid_io.hpp>
-#if defined(__linux__) || defined(__LINUX__)
+#if (defined(__linux__) || defined(__LINUX__)) && defined(ENABLE_WEBRTC_VIDEO)
 #include "video/WebRTCDecoder.h"
 #endif
 #include <slic3r/GUI/print_manage/AppUtils.hpp>
@@ -1304,8 +1304,8 @@ void PrinterMgrView::OnScriptMessage(wxWebViewEvent& evt)
             }
             else
             {
-                #if defined(__linux__) || defined(__LINUX__)
-                    WebRTCDecoder::GetInstance()->startPlay(video_url); 
+                #if (defined(__linux__) || defined(__LINUX__)) && defined(ENABLE_WEBRTC_VIDEO)
+                    WebRTCDecoder::GetInstance()->startPlay(video_url);
                 #endif
             }
 
diff --git a/src/slic3r/GUI/print_manage/App/SendToPrinter.cpp b/src/slic3r/GUI/print_manage/App/SendToPrinter.cpp
index 7e680785..ae8606ee 100644
--- a/src/slic3r/GUI/print_manage/App/SendToPrinter.cpp
+++ b/src/slic3r/GUI/print_manage/App/SendToPrinter.cpp
@@ -42,7 +42,7 @@
 #include <boost/uuid/uuid.hpp>
 #include <boost/uuid/uuid_generators.hpp>
 #include <boost/uuid/uuid_io.hpp>
-#if defined(__linux__) || defined(__LINUX__)
+#if (defined(__linux__) || defined(__LINUX__)) && defined(ENABLE_WEBRTC_VIDEO)
 #include "video/WebRTCDecoder.h"
 #endif
 #include "buildinfo.h"
@@ -417,7 +417,7 @@ void CxSentToPrinterDialog::OnScriptMessage(wxWebViewEvent& evt)
             }
             else
             {
-                #if defined(__linux__) || defined(__LINUX__)
+                #if (defined(__linux__) || defined(__LINUX__)) && defined(ENABLE_WEBRTC_VIDEO)
                     WebRTCDecoder::GetInstance()->stopPlay();
                 #endif
             }
-- 
2.53.0

